README.md

```diff
@@ -2,5 +2,435 @@
+        currency = Currency.getInstance(record.currency),
+        amount = record.amount.toULong(),
+        expireAt = record.expireAt,
+        entityState = entityState
+    )
+}
+```
+
+### Configure it
+We have unit of work and repository, so we can set up everything together.
+In this example we are going to use JDBC implementation for our transactional manager.
+```kotlin
+class WalletModule(databaseConfig: DatabaseConfig) {
+
+    /**
+     * Query executor definition
+     */
+    val transactionManager = JdbcTransactionManager(
+        primaryProvider = HikariPoolConnectionProvider(dataSource(databaseConfig, isPrimary = true)),
+        replicaProvider = HikariPoolConnectionProvider(dataSource(databaseConfig, isPrimary = false)),
+        blockingJdbcContext = newFixedThreadPool(databaseConfig.maxPoolSize.value()).asCoroutineDispatcher()
+    )
+    val queryExecutor = JdbcQueryExecutor(transactionManager)
+    val dslContext: DSLContext = DSL.using(
+        POSTGRES,
+        Settings().withRenderNamedParamPrefix("$").withParamType(ParamType.NAMED)
+    )
+
+    /**
+     * Persisting definition
+     */
+    val tracer = NoopTracerFactory.create()
+    val walletRepo = WalletRepository(queryExecutor, dslContext)
+    val persisting = Persisting(
+        transactionManager = transactionManager,
+        modelRepos = ModelRepos(Wallet::class hasRepo walletRepo),
+        eventRepository = JooqEventRepository(queryExecutor, dslContext, tracer)
+    )
+
+    /**
+     * Unit of work executor definition
+     */
+    val clock = Clock.tickMillis(UTC)
+    val uowx: UnitOfWorkExecutor = UnitOfWorkExecutor(
+        persisting = persisting,
+        tracer = tracer,
+        meterRegistry = SimpleMeterRegistry(),
+        factories = listOf(
+            CreateWalletUow::class withFactory { CreateWalletUow(walletRepo, clock) }
+        )
+    )
+}
+```
+
+### Run it!
+> Please don't forget to create tables for your models and table for storing events.
+You can find script to create event's table [here](eva-events-db-schema/src/main/resources/com/razz/eva/events/db)
+
+```kotlin
+    val module = WalletModule(config)
+    val principal = ServicePrincipal(Principal.Id("eva-id"), Principal.Name("eva"))
+    
+    val createdWallet = module.uowx.execute(CreateWalletUow::class, principal) {
+        CreateWalletUow.Params(
+            id = "45dfd599-4d62-47f1-8e47-a779df4f6bbc",
+            currency = "USD"
+        )
+    }
+```
+
+## Features
+
+### Event sourcing
+...
+
+### Unit of work validation
+After you wrote your first unit of work, you probably want to ask - how I can test it?
+
+We provide verification DSL, so you can write unit tests and verify results of your unit of work.
+Use `verifyInOrder` function to start verification process.
+```kotlin
+    CreateWalletUow(queries, clock).tryPerform(principal, params) verifyInOrder {
+        adds<Wallet> { model -> ... }
+        addsEq(expectedModel)
```

Comment:
Need clarification that `adds<M>` and `addsEq` are interchangeable. Form this snippet it's unclear that user has to provide either of this, not both.